plugin/core/sessions.py

```diff
@@ -603,7 +612,10 @@ def m_workspace_configuration(self, params: Dict[str, Any], request_id: Any) ->
         items = []  # type: List[Any]
         requested_items = params.get("items") or []
         for requested_item in requested_items:
-            items.append(self.config.settings.get(requested_item.get('section') or None))
+            configuration = self.config.settings.get(requested_item.get('section') or None)
+            if self._plugin:
+                self._plugin.on_workspace_configuration(requested_item, configuration)
+            items.append(configuration)
```

Comment:
The problem with this solution at the moment is that I have to access `_workspace_folders` from `on_workspace_configuration` handler, to use the "includes_uri" method. This is obviously a bit wrong so options are:
 1) make `_workspace_folders` public (remove the underscore)
 2) add some public method on session to do what `includes_uri` is doing
 3) pass the list of WorkspaceFolders to the handler

I think 3 is too specific at least.

Reply:
BTW. Can we treat all public methods of `Session` as something that plugins should be able to access? I imagine that could break easily.

Reply:
Having workspace_folders public is fine I think.

Reply:
I'll do that then but then I don't think we'll be able to keep the Session API fully stable...